internal/batches/service/workspaces.go

```diff
@@ -43,9 +43,14 @@ func findWorkspaces(
 	// Pre-compile all globs.
 	workspaceMatchers := make(map[batcheslib.WorkspaceConfiguration]glob.Glob)
 	for _, conf := range spec.Workspaces {
-		g, err := glob.Compile(conf.In)
+		in := conf.In
+		// Empty `in` should fall back to matching all, instead of nothing.
```

Comment:
Nitpick, but just wanted to mention it: we could also define something like

```go
type allMatcher struct{}
func (a allMatcher) Match(_ string) bool { return true }
```

and pass that in when `in == ""` since it satifies `glob.Glob`. We'd save a `glob.Compile`.